test/suites/RentingDappManager/CreateAgreement.test.ts

```diff
@@ -0,0 +1,384 @@
+            const productAll = parseLogs.map(ele => ele.args);
+
+            const productAddress = productAll[0][1];
+
+            //creating 5th agreement for this product
+            const productManagerInstanceESN = ProductManagerFactory.connect(
+                productAddress,
+                wallet2 ?? global.providerESN
+            );
+            
+            try {
+                const tx = await productManagerInstanceESN
+                    .connect(wallet2.connect(global.providerESN))
+                    .createAgreement(
+                        0,
+                        1060,
+                        1050
+                    );
+                    
+                ok(false, 'error should have been thrown, but not thrown');                
+            } catch (error) {
+                ok(
+                    error?.message?.includes('Not available on given range of timings'),
+                    'error is not related to overlapping timings'
+                );
+            }
+        });
+
+        it('tries to book an item (sixth time with incentive = 0 in available timings)', async () => {
+
+            //extracting product address from events log
+            const filter = global.rentingDappManagerInstanceESN.filters.ProductDetails(wallet1.address,null,null,null,null,null,null,null,null,null);
+            const logs = await global.rentingDappManagerInstanceESN.queryFilter(filter);
+            const parseLogs = logs.map((log) => global.rentingDappManagerInstanceESN.interface.parseLog(log));
+            const productAll = parseLogs.map(ele => ele.args);
+
+            const productAddress = productAll[0][1];
+
+            //creating 6th agreement for this product
+            const productManagerInstanceESN = ProductManagerFactory.connect(
+                productAddress,
+                wallet2 ?? global.providerESN
+            );
+
+            const receipt = await parseReceipt(
+                productManagerInstanceESN
+                    .connect(wallet2.connect(global.providerESN))
+                    .createAgreement(
+                        0,
+                        1062,
+                        1150
+                    ),
+                //true
+            );
+
+            const parsedLogs = receipt.logs.map((log) =>
+                productManagerInstanceESN.interface.parseLog(log)
+            );
+
+            ok(
+                parsedLogs[0].args[0] === wallet1.address,
+                "Lessor should match wallet1 address"
+            );
+
+            ok(
+                parsedLogs[0].args[1] === wallet2.address,
+                "Lessee should match wallet2 address"
+            );
+
+            ok(
+                ethers.utils.isHexString(parsedLogs[0].args[2]),
+                'Rental agreement address added should be hexstring'
+            );
+        });
+
+
+        it('tries to book an item (incentive = 0 in available timings)', async () => {
```

Comment:
"tries" means it is going to error. From the test logic it looks it expects success. You can use a following pattern for the sake of consistency with test case descriptions of other contracts

```suggestion
        it('books an item (incentive = 0 in available timings)', async () => {
```

Reply:
Also, as in previous comment, there are similar descriptions if you can add a differentiator, it'd be great!

Reply:
Sone